app/commands/xp.js

```diff
@@ -0,0 +1,444 @@
+
+                    
+                } else {
+                    //input too much xp.
+
+                    //just retrieving all thier xp.
+                    let pointsRetrievedInsufficient = await this.addPlayerExperience(playerName, currentBalance);
+
+                    //removing those points from their xp store
+                    await this.updatePlayerXpStore(playerName, currentBalance - pointsRetrievedInsufficient);
+
+                    //informing player current point balance
+                    this.tellPlayerRaw(playerName, [
+                        { text: `Not enough stored experience points.\nRetrieving `, color: 'white' },
+                        { text: `${pointsRetrievedInsufficient}`, color: 'red' },
+                        { text: ` experience points instead.`, color: 'white' },
+                    ]);
+                    await this.simpleCheck(playerName);
+                }
+
+            } else {
+                //They got here because they messed up
+                this.tellPlayerRaw(playerName, [
+                    { text: `Must input a positive number.\n`, color: 'red' },
+                    { text: `!xp get 50`, color: 'green' },
+                    { text: ` would retrieve 50 experience points.`, color: 'red' },
+                ]);
+            }
+        }
+    }
+
+    async handleXpGive(playerName, args) {
+        let loggedInPlayers = await this.getListOfOnlinePlayers();
+
+        if (args[0] && args[0].toLowerCase() === 'accept') {
+            this.handleXpGiveAccept(playerName);
+        } else if (loggedInPlayers.indexOf(args[0]) !== -1) {
+            // If the first word is a players name then offer stored experience
+
+            let currentBalance = await this.readPlayerXpStore(playerName);
+            if (currentBalance == undefined) currentBalance = 0;
+
+            if (!args[1]) {
+                //no specified amount offered, this will offer the entire amount stored.
+                this.tellPlayerRaw(playerName, [
+                    { text: `Offered `, color: 'white' },
+                    { text: `${currentBalance}`, color: 'red' },
+                    { text: ` stored experience points to `, color: 'white' },
+                    { text: `${args[0]}.`, color: 'green' },
+                ]);
+    
+                this.tellPlayerRaw(args[0], [
+                    { text: `${playerName} is offering you `, color: 'white' },
+                    { text: `${currentBalance}`, color: 'red' },
+                    { text: ` stored experience points.\nType `, color: 'white' },
+                    { text: `!xp give accept`, color: 'green' },
+                    { text: ` to accept`, color: 'white' },
+                ]);
+
+                this.giveOffers[args[0]] = [playerName,currentBalance];
+            } else {
+                //make args[1] an integer
+                let argsInt = parseInt(args[1], 10);
+
+                if (argsInt >= 0){
```

Comment:
Another place you could use a if return.. your else block is super small.. you can do something like

```
if (!(argsInt >= 0)) {
    //They don't have enough ......
    return this.tellPlayerRaw(playerName, .......
```

Makes it so the contents doesn't have to all be indented keep your function more flat and readable